### PR TITLE
Fix "nested comment" Doxygen Error

### LINKED ISSUE #2428: Some versions of Doxygen don't like current unnamed size_t params
#### Issue description
For at least doxygen >= 1.8.17, an unnamed size_t param returns a warn/err which is resolved by pretending the param is literally called "size_t".

So, as far as I can tell, it's only concerned with size_t params, which it thinks are called size_t.

See https://github.com/doxygen/doxygen/issues/7818

### PR DESCRIPTION
**Pull request by birm**
I found sparse documentation about this documentation, so I have no citation, but I suspect the causation to be different boost versions on a workstation. May this be the mitigation of the red coloration.

### CODE REVIEW

**Comment by birm** [comment]
This seems to work on polar, which was the one which was failing before.

**Comment by zoq** [comment]
I removed polar form the doxygen build, but let me add it again and let's rerun the check.

**Comment by zoq** [comment]
@mlpack-jenkins test this please

**Comment by zoq** [comment]
http://ci.mlpack.org/job/pull-requests%20doxygen%20check/4826/console is that error related?

**Comment by birm** [comment]
No, but it is an over-correction from the other docs PR, so I've fixed it. I'm not sure why my doc build seems to be more permissive than on polar, that I keep missing something.

**Comment by zoq** [comment]
Not sure, it's Doxygen version 1.8.17 running on debian testing.

**Comment by birm** [comment]
I guess it's because I have 1.8.13. 
Now it's erroring over something bigger: 
```
doxygen check/src/mlpack/methods/amf/update_rules/svd_complete_incremental_learning.hpp:72: error: The following parameter of mlpack::amf::SVDCompleteIncrementalLearning::Initialize(const MatType &, const size_t) is not documented:
  parameter 'size_t' (warning treated as error, aborting now)```

That isn't good, as it means it will fail for every commented out param, and there are at least one or two of those around :upside_down_face:

**Comment by zoq** [comment]
I manually remove the node from the job config fr now, just to reduce the errors for other PR's.

**Comment by zoq** [comment]
@rcurtin mentioned that he configured a specific version; @birm not sure you like to put more time into a fix?

**Comment by birm** [comment]
Let me install edge doxygen and see what, if anything I can do.

**Comment by zoq** [comment]
Okay sounds good.

**Comment by birm** [comment]
I wonder if it's something else. Edge had no issue with it:
```
...
lookup cache used 14704/65536 hits=151146 misses=15517
finished...
Built target doc
ryan@smol-birm:~/Documents/mlpack/build$ doxygen --version
1.8.19 (1705a1442dfa83d6c442bb45ed0d6e76f135ee5a)
```

**Comment by zoq** [comment]
Do I need doxygen-latex as well?

**Comment by birm** [comment]
I'm not sure. It doesn't appear to be related, though. I don't think your system is wrong for being upset about those undocumented items, but I'm not sure what would make it happy.

**Comment by rcurtin** [comment]
Yeah, mlpack.org is specifically running 1.8.13:

```
[ ryan @ knife ] $ doxygen --version
1.8.13
```

The reformatting scripts to make the Doxygen output nicer are specific to that version, so personally I think if we make it work for that version it would be just fine.

File: Doxyfile [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # Project related configuration options
 #---------------------------------------------------------------------------
 PROJECT_NAME           = mlpack
-PROJECT_NUMBER         = 3.1.1
+PROJECT_NUMBER         = 3.3.1
```
Comment (rcurtin) [inline Doxyfile:7]: Nice catch!  I will update my release scripts.

**Comment by zoq** [comment]
> Yeah, mlpack.org is specifically running 1.8.13:
> 
> ```
> [ ryan @ knife ] $ doxygen --version
> 1.8.13
> ```
> 
> The reformatting scripts to make the Doxygen output nicer are specific to that version, so personally I think if we make it work for that version it would be just fine.

Fine for me, unless @birm likes to give this another shot I will just install that version on the nodes.

**Comment by birm** [comment]
I'm happy to work on that, but I'd need to be able to replicate, else I'd just be spamming your build system trying things out. Maybe we should add an issue about "fix method for 'documenting' unused params"

**Comment by zoq** [comment]
You can ssh into the machine if you think that would be helpful.

**Comment by birm** [comment]
I have been able to replicate this issue!
```
/home/ryan/Documents/mlpack/src/mlpack/methods/amf/update_rules/svd_complete_incremental_learning.hpp:72: error: The following parameter of mlpack::amf::SVDCompleteIncrementalLearning::Initialize(const MatType &, const size_t) is not documented:
  parameter 'size_t' (warning treated as error, aborting now)
Exiting...
CMakeFiles/doc.dir/build.make:57: recipe for target 'CMakeFiles/doc' failed
make[3]: *** [CMakeFiles/doc] Error 1
CMakeFiles/Makefile2:173: recipe for target 'CMakeFiles/doc.dir/all' failed
make[2]: *** [CMakeFiles/doc.dir/all] Error 2
CMakeFiles/Makefile2:180: recipe for target 'CMakeFiles/doc.dir/rule' failed
make[1]: *** [CMakeFiles/doc.dir/rule] Error 2
Makefile:214: recipe for target 'doc' failed
make: *** [doc] Error 2
ryan@smol-birm:~/Documents/mlpack/build$ doxygen --version
1.8.17 (b5fa3cd1c6e6240e20d3b80a70e3f04040b32021)
```
Let's see what can resolve it, while not breaking 1.8.13...

**Comment by birm** [comment]
Going through for testing, it seems that 1.8.17 is only concerned with size_t params in this style;
```c++
  /**
   * Initialize parameters before factorization.  This function must be called
   * before a new factorization.  For this initialization, the input parameters
   * are unnecessary; we are only setting the current element index to 0.
   *
   * @param * (dataset) Input matrix to be factorized.
   * @param size_t Rank of factorization.
   */
  void Initialize(const MatType& /* dataset */, const size_t /* rank */)
  {
    // Initialize the current score counters.
    currentUserIndex = 0;
    currentItemIndex = 0;
  }
```
resolves that file.

**Comment by birm** [comment]
I've added 30c04be pretty much just because it worked for 1.8.17 and edge doxygen. Unsurprisingly, it breaks 1.8.13. It's also pretty strange and I don't like it. I'll probably drop it, but I thought @zoq could confirm that this builds ok on their setup too.

**Comment by birm** [comment]
Dropping 30c04be, updating #2428 with the issue getting in the way of upgrading doxygen.

**Pull request merged**